clients/src/main/java/org/apache/kafka/common/security/plain/PlainSaslServer.java

```diff
@@ -93,11 +94,11 @@ public PlainSaslServer(JaasContext jaasContext) {
         String expectedPassword = jaasContext.configEntryOption(JAAS_USER_PREFIX + username,
                 PlainLoginModule.class.getName());
         if (!password.equals(expectedPassword)) {
-            throw new SaslException("Authentication failed: Invalid username or password");
+            throw new AuthenticationException("Authentication failed: Invalid username or password");
```

Comment:
Don't we have to update `ScramSaslServer` as well?

Reply:
I left the default message for SCRAM since clients dont exactly provide username/password in the SCRAM message and we dont want to say which check failed. I have updated the message when the authorization id requested is not the username.